src/renderer/components/RuntimeAdapter.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useEffect } from 'react'
+
+import useMessage from '../hooks/useMessage'
+import useProcessQr from '../hooks/useProcessQr'
+import { clearNotificationsForChat } from '../system-integration/notifications'
+import { runtime } from '../runtime'
+import { selectedAccountId } from '../ScreenController'
+
+import type { PropsWithChildren } from 'react'
+
+/**
+ * Helper component to hook React methods into the external "runtime". This
+ * allows us to interact with the underlying Electron runtime and
+ * operating system.
+ */
+export default function RuntimeAdapter({ children }: PropsWithChildren<{}>) {
+  const processQr = useProcessQr()
+  const { jumpToMessage } = useMessage()
+
+  useEffect(() => {
+    runtime.onOpenQrUrl = (url: string) => {
+      const accountId = selectedAccountId()
+      processQr(accountId, url)
+    }
+
+    runtime.setNotificationCallback(async ({ accountId, msgId, chatId }) => {
+      if (window.__selectedAccountId !== accountId) {
+        await window.__selectAccount(accountId)
+      }
```

Comment:
All of this accountId stuff smells a little bit, but I think it's fine

Reply:
can't you also pass the account id and account selection function to this component?

Reply:
I mean why is the hook passed:
1. in `MessageListView.tsx` to `withSendMessageHook(MapComponent)` as prop
2. in `withSendMessageHook` it is not used from the prop instead fetched from the context again and then set to the `MapComponent` as prop. technically `MapComponent` gets `sendMessage` set two times or am I missing sth?